libs/clients/althingi-ombudsman/src/lib/client-althingi-ombudsman.middleware.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@nestjs/common'
+import {
+  FetchParams,
+  Middleware,
+  RequestContext,
+  SecurityApi,
+} from '../gen/fetch/dev'
+
+@Injectable()
+export class TokenMiddleware implements Middleware {
+  constructor(
+    private readonly password: string,
+    private readonly username: string,
+    private readonly securityApi: SecurityApi,
+  ) {}
+
+  async pre(context: RequestContext): Promise<FetchParams | void> {
```

Comment:
So this will be triggered before each request.
What is the lifetime of the token returned by the securityApi? Can it be reused while it is still valid?

Reply:
The token should be valid for 7 days however I think that is a bit long for caching. I'll cache it for 1 day to reduce unnecessary authentication calls.